packages/qwik-city/runtime/src/spa-shim.ts

```diff
@@ -33,18 +31,15 @@ const shim = async (path: string, symbol: string) => {
     }
 
     const currentScript = document.currentScript as HTMLScriptElement;
-    if (!isDev) {
-      (await import(path))[symbol](currentScript);
-    } else {
-      // Importing @qwik-city-plan here explodes dev, get basePathname manually.
-      const container = currentScript.closest('[q\\:container]')!;
-      const base = new URL(container.getAttribute('q:base')!, document.baseURI);
-      const url = new URL(path, base);
-
-      // Bypass dev import hijack. (not going to work here)
-      // eslint-disable-next-line no-new-func
-      const imp = new Function('url', 'return import(url)');
-      (await imp(url.href))[symbol](currentScript);
-    }
+
+    // Importing @qwik-city-plan here explodes dev, get basePathname manually.
+    const container = currentScript.closest('[q\\:container]')!;
+    const base = new URL(container.getAttribute('q:base')!, document.baseURI);
+    const url = new URL(path, base);
+
+    // Bypass dev import hijack. (not going to work here)
+    // eslint-disable-next-line no-new-func
+    const imp = new Function('url', 'return import(url)');
```

Comment:
seems like this hack is needed for dev mode only. So I think we still in `if (isDev) {}`